packages/components/src/lib/d3/patchedScales.ts

```diff
@@ -0,0 +1,97 @@
+import * as d3 from "d3";
+
+type CustomFormat = "squiggle-default" | undefined;
+
+// see lib/d3/index.ts
+export const defaultTickFormatSpecifier: CustomFormat = "squiggle-default";
+
+function isCustomFormat(
+  specifier: string | undefined
+): specifier is CustomFormat {
+  return specifier === "squiggle-default" || specifier === undefined;
+}
+
+function squiggleDefaultFormat() {
+  const locale = d3.formatLocale({
+    decimal: ".",
+    thousands: ",",
+    grouping: [3],
+    currency: ["$", ""],
+    minus: "-",
+  });
+  const siFormat = locale.format(".9~s");
+  const expFormat = locale.format(".9~e");
+  const fixedFormat = locale.format(".9~f");
+
+  return (d: d3.NumberValue) => {
+    const abs = Math.abs(Number(d));
+    if (abs === 0 || (abs >= 0.0001 && abs < 1e6)) {
+      return fixedFormat(d);
+    } else if (abs >= 1e9 && abs < 1e12) {
+      return fixedFormat(Number(d) / 1e9) + "B";
+    } else if (abs >= 1e6 && abs < 1e15) {
```

Comment:
Is the 1e6 part intentional? My hunch would just be to make a different clause above, for readability. Like,
```else if (abs < 1e9)...```

Reply:
For that matter, I also find line 28 awkward. I just want this to go in order, even if it means repeating some of the formats.